### PR TITLE
few tweaks for objects api

### PR DESCRIPTION
**Pull request by Xavrax**
feat: added `filter` and `sort` parameters to be closer to the other SDKs with object API. 

added `filter` and `sort` parameters to be closer to the other SDKs with object API. 

feat: configurable `bool` type.

configurable `bool` type.

fix: missing features needed for grant token API in CMakeLIsts.txt

missing features needed for grant token API in CMakeLIsts.txt

### CODE REVIEW

**Comment by Xavrax** [comment]
Additionally, added possibility to add suffix for `pnsdk` parameter

File: core/pubnub_objects_api.h [hunk 1]
```diff
@@ -8,9 +8,118 @@
 
 #include "pubnub_api_types.h"
 #include "lib/pb_extern.h"
+#include "stddef.h"
```
Comment (parfeon) [inline core/pubnub_objects_api.h:11]: @Xavrax what is this import? Where did it come from?

File: core/samples/pubnub_objects_api_sample.c [hunk 2]
```diff
@@ -0,0 +1,473 @@
+#include "pubnub_api_types.h"
+#include "pubnub_blocking_io.h"
+#include "pubnub_pubsubapi.h"
+#include "core/pubnub_helper.h"
+#include "core/pubnub_alloc.h"
+#include "core/pubnub_ntf_sync.h"
+#include "core/pubnub_objects_api.h"
+#include <stdio.h>
+#include <unistd.h>
```
Comment (parfeon) [inline core/samples/pubnub_objects_api_sample.c:9]: It looks like this header from `cgreen` - do we really need it?

**Review by parfeon — CHANGES_REQUESTED** [review]
There are some questions. Maybe @KGronek share his idea.

File: core/samples/pubnub_objects_api_sample.c (hunk 2 above)
Comment (parfeon) [reply core/samples/pubnub_objects_api_sample.c:9]: I see, it is for `sleep`.

File: core/pubnub_objects_api.h (hunk 1 above)
Comment (parfeon) [reply core/pubnub_objects_api.h:11]: IDE shows that `include` not used.

**Review by parfeon — APPROVED** [review]
LGTM!

**Comment by parfeon** [comment]
@pubnub-release-bot release

**Pull request merged**